### DISCUSSION TITLE
Should we change `auto_session_enable_last_session` option to false

### DISCUSSION THREAD

**Discussion by Saafo** [ayamir/nvimdots#704] [Polls]
1. The `auto-session` plugin has a `auto_session_enable_last_session` option, which is described here: https://github.com/rmagatti/auto-session#last-session

I think the default value `false` is more suitable for more people since no one want one workspace's config(especially **buffers**) to be appeared in another workspace. The value is currently `true`:

https://github.com/ayamir/nvimdots/blob/3be4c9bc862ce8f67e4ce712cff893b96d215754/lua/modules/configs/editor/auto-session.lua#L4

2. Another thing quite odd is that `auto-session` is currently not enabled since `lazy = true` and no event is configured. Does this due to no one is using this plugin?

https://github.com/ayamir/nvimdots/blob/3be4c9bc862ce8f67e4ce712cff893b96d215754/lua/modules/plugins/editor.lua#L9

**Comment by ayamir** [comment]
Personally I don't use session function.

**Reply by charliie-dev** [reply]
true, use `tmux` or `wezterm` is muscle memory to me now.

**Reply by charliie-dev** [reply]
I start playing with `zellij` recently. `zellij` is basically `tmux` but in rust.
However, `zullij` is lack of collaboration plugins with nvim. It's still very new comparing to `tmux`.

**Reply by Saafo** [reply]
> true, use tmux or wezterm is muscle memory to me now.

So you mean you do not open multiple tabs in nvim? Just open multiple terminals by tmux and each open a vim file directly??

**Reply by charliie-dev** [reply]
> So you mean you do not open multiple tabs in nvim?

`bufferline` is enough for me to edit multiple files when working in small or medium size project.
I rarely use `tab` in nvim, as `bufferline` will also "unfold" the nvim tabs.

For a huge monorepo project, I would prefer useing tmux's session/windows or wezterm's workspace to only focus on single point and clear other noises.

I also like tmux for it can be put to background without needing to keep the ssh connections alive. This become extremely useful when doing some time-consuming tasks, like training a ML model or compiling a huge project. 

TBH, I lives in `terminal`, not in `nvim`. I like using cli tools, nvim is just parts of it. I prefer a pure terminal than a nvim terminal for it just works smoother.

Multiple nvim instances does have it's drawback, like higher memory usage. But I got 32GB/64GB of ram, so who cares lol

**Reply by Saafo** [reply]
I'm sorry because I've confused tabs with bufferline. What I said `tabs` should be considered as `bufferline` 🤣.

So don't you have the need to remember the buffers you've opened in a directory?

**Reply by charliie-dev** [reply]
Actually no, I can mostly remember the project structure for a small/medium projects. So I just quit nvim and cd to the desired folder, use `<leader>ff` to open files in a project, or use `nvim-tree` to open other files under same level of directory.

side note: I use `zoxide` to do super jumps between different level of folders and projects. So it's relatively quick for me to just use `cd`(actually it's `z` for `zoxide`).

**Reply by Saafo** [reply]
Got it!

**Reply by charliie-dev** [reply]
@Saafo u can check my [dotfiles repo](https://github.com/CharlesChiuGit/dotfiles) if u r interested.
it will always be WIP, since I just keep moving things around lol.

**Comment by Saafo** [comment]
Since no one is using auto-session, can I just change `auto_session_enable_last_session` to false(by deleting it)?

**Reply by charliie-dev** [reply]
let's wait for a couple days. it's too early to set the tone.

**Reply by Saafo** [reply]
You're correct, I've found that auto-session seriously slow down startup time. I'm currently finding workaround solutions or alternative plugins to recover buffers.

Edit: found the reason and waiting for fix up: https://github.com/rmagatti/auto-session/issues/219
